File: 643. Maximum Average Subarray I/Solution.cs
public class Solution {
    public double FindMaxAverage(int[] nums, int k) {
        
        int sumInsideWindow = nums.Take(k).Sum();
        int maxSumInsideWindow = sumInsideWindow;

        for (int i = k; i < nums.Length; ++i) {
            sumInsideWindow += (nums[i]-nums[i-k]);
            maxSumInsideWindow = Math.Max(maxSumInsideWindow, sumInsideWindow);
        }

        return maxSumInsideWindow / (double)k;
    }
}

File: 643. Maximum Average Subarray I/solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

        sumInsideWindow = sum(nums[0:k])
        maxSumInsideWindow = sumInsideWindow

        for i in range(k,len(nums)):
            sumInsideWindow += (nums[i]-nums[i-k])
            maxSumInsideWindow = max(maxSumInsideWindow, sumInsideWindow)

        return maxSumInsideWindow / k
